chimney-macro-commons/src/main/scala-3/io/scalaland/chimney/internal/compiletime/ExprPromisesPlatform.scala

```diff
@@ -0,0 +1,144 @@
+package io.scalaland.chimney.internal.compiletime
+
+private[compiletime] trait ExprPromisesPlatform extends ExprPromises { this: DefinitionsPlatform =>
+
+  import quotes.*, quotes.reflect.*
+
+  final override protected type ExprPromiseName = Symbol
+
+  protected object ExprPromise extends ExprPromiseModule {
+
+    // made public for ProductType.parse
+    def provideFreshName[From: Type](
+        nameGenerationStrategy: NameGenerationStrategy,
+        usageHint: UsageHint
+    ): ExprPromiseName =
+      nameGenerationStrategy match {
+        case NameGenerationStrategy.FromPrefix(src) => freshTermName[From](src, usageHint)
+        case NameGenerationStrategy.FromType        => freshTermName[From](usageHint)
+        case NameGenerationStrategy.FromExpr(expr)  => freshTermName[From](expr, usageHint)
+      }
+
+    protected def createRefToName[From: Type](name: ExprPromiseName): Expr[From] = Ref(name).asExprOf[From]
+
+    def createLambda[From: Type, To: Type, B](
+        fromName: ExprPromiseName,
+        to: Expr[To]
+    ): Expr[From => To] = '{ (param: From) =>
+      ${
+        PrependDefinitionsTo.initializeDefns[To](
+          vals = Vector((fromName, ExistentialExpr('{ param }), PrependDefinitionsTo.DefnType.Val)),
+          expr = to
+        )
+      }
+    }
+
+    def createLambda2[From: Type, From2: Type, To: Type, B](
+        fromName: ExprPromiseName,
+        from2Name: ExprPromiseName,
+        to: Expr[To]
+    ): Expr[(From, From2) => To] = '{ (param: From, param2: From2) =>
+      ${
+        PrependDefinitionsTo.initializeDefns[To](
+          vals = Vector(
+            (fromName, ExistentialExpr('{ param }), PrependDefinitionsTo.DefnType.Val),
+            (from2Name, ExistentialExpr('{ param2 }), PrependDefinitionsTo.DefnType.Val)
+          ),
+          expr = to
+        )
+      }
+    }
+
+    private def freshTermName[A: Type](prefix: String, usageHint: UsageHint): ExprPromiseName = Symbol.newVal(
+      Symbol.spliceOwner,
+      FreshTerm.generate(prefix),
+      TypeRepr.of[A],
+      usageHint match
+        case UsageHint.None => Flags.EmptyFlags
+        case UsageHint.Lazy => Flags.Lazy
+        case UsageHint.Var  => Flags.Mutable
+      ,
+      Symbol.noSymbol
+    )
+    private def freshTermName[A: Type](usageHint: UsageHint): ExprPromiseName =
+      freshTermName(TypeRepr.of[A].show(using Printer.TypeReprShortCode).toLowerCase, usageHint)
+    private def freshTermName[A: Type](expr: Expr[?], usageHint: UsageHint): ExprPromiseName =
+      freshTermName[A](toFieldName(expr), usageHint)
+
+    // Undoes the encoding of freshTermName so that generated value would not contain $1, $2, ... and weird
+    // dot-replacement - this makes generated fresh names more readable as it prevents e.g. typename$macro$1$2$3
+    private def toFieldName[A](expr: Expr[A]): String =
+      if expr.asTerm.toString.contains("$u002E") then {
+        println(expr.asTerm.toString)
```

Comment:
Looks like unfinished debugging. What should be the final version?

Reply:
It can be removed - I was checking which workarounds from Scala 2 are necessary and which could be removed and this checked for something that could be removed but I hadn't removed this println.

Reply:
Removed